test/curry.js

```diff
@@ -119,6 +119,15 @@ describe('curry', function() {
     eq(g(1)(2)(3, 4), [1, 2, 3, 4]);
   });
 
+  it('return value when collet value which is not placeholder', function() {
+    var f = function(a, b, c) { return [a, b, c]; };
+    var g = R.curry(f);
+    var _ = R.__;
+
+    eq(g(1)(_)(2, _, _)(3), [1, 2, 3]);
+    eq(g(1)(_, _, _)(2, _, _)(3), [1, 2, 3]);
+    eq(g(1)(_, _, _, 4)(2, _, _)(3), [1, 2, 3]);
```

Comment:
Here are some more tests we can add

```
    eq(g(_, 2, 3, 4)(1), [1, 2, 3]);
    eq(g(1, 2, 3, _), [1, 2, 3]);
    eq(g(1)(_, 3, 4)(2), [1, 2, 3]);
    eq(g(_)(_, 2, 3, 4)(1), [1, 2, 3]);
    eq(g(_)(_, 2, _, 4, 5)(_, 3, 4)(1), [1, 2, 3]);
    eq(g(_)(_, 2, _, 4, 5)(1, 3), [1, 2, 3]);
```

Reply:
`g(_)(_, 2, _, 4, 5)(_, 3, 4)(1)`  This case will cause `_arity` get negative integer value, and the reason is combined get value more than `length` cause left < 0 
![image](https://github.com/ramda/ramda/assets/54520846/b6e057f2-e8e9-48ac-848f-a44adac4b565)
and I don't not how to handle it.

Reply:
Shoulid i limt `left` min value ?

Reply:
ah I had another change I didn't mention locally to account for that.. but thinking about it some more maybe a better solution is:
```js
      if (combinedIdx < length) {
        if (!_isPlaceholder(result)) {
          left -= 1;
        }
        else {
          hasPlaceholderBeforeFinalArg = true;
        }
      }
```
We only care to update both `left` and `hasPlaceholderBeforeFinalArg` if we are less then than arity (`length`), so I think this should handle it? All our tests passed for me locally